plasmapy/formulary/tests/test_collisions.py

```diff
@@ -1094,13 +1094,40 @@ def test_handle_nparrays(self, insert_some_nans, insert_all_nans, kwargs):
             impact_parameter, insert_some_nans, insert_all_nans, kwargs
         )
 
-    def test_extend_scalar_bmin(self):
+    @pytest.mark.parametrize(
+        "n_e_shape,T_shape",
+        # Scalar T
+        [
+            ((2, 3, 5), (1,)),
+            # Scalar n
+            ((1,), (2, 3, 5)),
+            # Both arrays of equal size
+            ((2, 3, 5), (2, 3, 5)),
+            # Higher dimensional test
+            ((2, 3, 5, 4, 2), (2, 3, 5, 4, 2)),
+        ],
+    )
+    def test_extend_output_for_array_input(self, n_e_shape, T_shape):
         """
-        Test to verify that if T is scalar and n is vector, bmin will be extended
-        to the same length as bmax
+        Test to verify that if either/or T and n_e are arrays, the resulting
+        bmin and bmax have the correct shapes.
+
+        This is necessary in addition to test_handle_nparrays to ensure that
+        the output arrays are extended correctly.
+
         """
-        (bmin, bmax) = impact_parameter(1 * u.eV, self.n_e_arr, self.particles)
-        assert len(bmin) == len(bmax)
+
+        output_shape = max(map(len, [T_shape, n_e_shape]))
```

Comment:
```suggestion
        output_shape = T_shape if len(T_shape) >= len(n_e_shape) else n_e_shape
```
Oops!  Let's try the other option...